### ISSUE TITLE
Cannot use APOC aggregation function in subquery

### ISSUE THREAD

**Issue by rickardoberg** [neo4j/neo4j#13322]
**Neo4j version**: 5.12.0
**Operating system**: Windows 11
**API/Driver**: Cypher

### Steps to reproduce
```
CALL { 
    UNWIND [1,2,3] as numbers 
    WITH numbers
    RETURN { max: max(numbers),median: apoc.agg.median(numbers)} as values
}
RETURN values
```
### Expected behavior
{max:3,median:2}
### Actual behavior
Error message:

> Aggregation column contains implicit grouping expressions. For example, in 'RETURN n.a, n.a + n.b + count(*)' the aggregation expression 'n.a + n.b + count(*)' includes the implicit grouping key 'n.b'. It may be possible to rewrite the query by extracting these grouping/aggregation expressions into a preceding WITH clause. Illegal expression(s): numbers (line 4, column 56 (offset: 111))
> "    RETURN { max: max(numbers),median: apoc.agg.median(numbers)} as values"
> 

This only seems to happen with non-builtin functions. If you replace median with "avg" it works as expected.

**Comment by hedengran** [comment]
Hi @rickardoberg , thank you for the report. We are looking into it, and will get back to you.

**Comment by gem-neo4j** [comment]
Hi! As a workaround in the meantime, you can do the following:

```
CALL { 
    UNWIND [1,2,3] as numbers 
    WITH max(numbers) as max, apoc.agg.median(numbers) as median
    RETURN { max: max, median: median} as values
}
RETURN values
```

**Comment by gem-neo4j** [comment]
Thanks for reporting, this is fixed and will be available in the 5.14 release in a month :)
